app/controllers/AtomActions.scala

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import cats.syntax.either._
+import com.gu.contentatom.thrift.{Atom, EventType}
+import db.AtomDataStores._
+import db.AtomWorkshopDBAPI
+import models._
+import play.api.libs.ws.WSClient
+import play.api.mvc.Controller
+import services.AtomPublishers._
+import services.NotificationLists
+import util.AtomLogic._
+import util.AtomUpdateOperations._
+import io.circe._
+import com.gu.fezziwig.CirceScroogeMacros._
+import com.gu.pandomainauth.action.UserRequest
+
+class AtomActions(val wsClient: WSClient, val atomWorkshopDB: AtomWorkshopDBAPI) extends Controller with PanDomainAuthActions {
+
+  def createNotificationList(atomType: String, id: String) = AuthAction { req =>
+    APIResponse {
+      for {
+        currentDraftAtom <- getCurrentDraftAtom(atomType, id)
+        postHookAtom <- NotificationLists.createNotificationList(currentDraftAtom)
+        updatedAtom <- publishUpdatedAtom(postHookAtom, req)
+      } yield updatedAtom
+    }
+  }
+
+  def deleteNotificationList(atomType: String, id: String) = AuthAction { req =>
+    APIResponse {
+      for {
+        currentDraftAtom <- getCurrentDraftAtom(atomType, id)
+        postHookAtom <- NotificationLists.deleteNotificationList(currentDraftAtom)
+        updatedAtom <- publishUpdatedAtom(postHookAtom, req)
+      } yield updatedAtom
+    }
+  }
+
+  private def getCurrentDraftAtom(atomType: String, id: String): Either[AtomAPIError, Atom] = for {
+    atomType <- validateAtomType(atomType)
+    currentDraftAtom <- atomWorkshopDB.getAtom(previewDataStore, atomType, id)
+  } yield currentDraftAtom
+
+  private def publishUpdatedAtom(atom: Atom, req: UserRequest[_]): Either[AtomAPIError, Atom] = for {
+    updatedAtom <- atomWorkshopDB.publishAtom(publishedDataStore, req.user, updateTopLevelFields(atom, req.user, publish=true))
+    _ <- atomWorkshopDB.updateAtom(previewDataStore, updatedAtom)
```

Comment:
Should these three actions occur sequentially? I wonder if at least the two below could be triggered at the same time and closed on together; what do you think?

Reply:
good point, certainly the last 2 can